builtin/providers/aws/resource_aws_instance.go

```diff
@@ -986,7 +997,15 @@ func buildAwsInstanceOpts(
 		}
 	}
 
-	if hasSubnet && associatePublicIPAddress {
+	if hasNetworkInterface {
+		ni := &ec2.InstanceNetworkInterfaceSpecification{
+			NetworkInterfaceId: aws.String(networkInterfaceID),
+			DeviceIndex:        aws.Int64(int64(0)),
+			SubnetId:           aws.String(subnetID),
```

Comment:
We're also don't yet consider `private_ip` or `vpc_security_group_ids`, are those coming, or do they not apply here?